test/test_filesystem.cpp

```diff
@@ -330,22 +360,25 @@ TEST_F(TestFilesystemFixture, calculate_directory_size) {
 #ifdef WIN32
   // Due to different line breaks on windows, we have one more byte in the file.
   // See https://github.com/ros2/rcutils/issues/198
-  ASSERT_EQ(6u, size);
+  EXPECT_EQ(6u, size);
 #else
-  ASSERT_EQ(5u, size);
+  EXPECT_EQ(5u, size);
 #endif
-  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
-  {
-    g_allocator.deallocate(path, g_allocator.state);
-  });
+  g_allocator.deallocate(path, g_allocator.state);
 
   char * non_existing_path = rcutils_join_path(this->test_path, "non_existing_folder", g_allocator);
   size = rcutils_calculate_directory_size(non_existing_path, g_allocator);
-  ASSERT_EQ(0u, size);
-  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+  EXPECT_EQ(0u, size);
+  g_allocator.deallocate(non_existing_path, g_allocator.state);
+
   {
-    g_allocator.deallocate(non_existing_path, g_allocator.state);
-  });
+    auto fs = mocking_utils::patch_filesystem("lib:rcutils");
+    const char * path = "some_fake_directory/some_fake_folder";
+    fs.file_info(path).st_mode |= mocking_utils::filesystem::FileTypes::DIRECTORY;
+    fs.exhaust_file_descriptors();
+    size = rcutils_calculate_directory_size(path, g_allocator);
+    EXPECT_EQ(0u, size);
```

Comment:
It could, but I'd keep changes to `rcutils` implementation out of this PR. [Same as before](https://github.com/ros2/rcutils/pull/272#discussion_r463206164).

Reply:
Sounds good, I can help you opening an enhancement issue targeting that if you'd like.